src/pipert2/core/handlers/message_handler.py

```diff
@@ -1,5 +1,52 @@
 from abc import ABC, abstractmethod
+from pipert2.core.base.message import Message
 
 
 class MessageHandler(ABC):
-    pass
+    """The message handler is responsible for receiving messages
+    and relaying them onward if needed.
+    The message handler communicates only with the network object
+    and not with other routines directly.
+    """
+
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def _get(self) -> Message:
```

Comment:
Not sure if the Message type hints should be here if you return bytes of the object